packages/flutter_blue_plus_platform_interface/lib/src/bluetooth_msgs.dart

```diff
@@ -515,6 +529,7 @@ class BmReadDescriptorRequest {
     data['remote_id'] = remoteId.str;
     data['service_uuid'] = serviceUuid.str;
     data['characteristic_uuid'] = characteristicUuid.str;
+    data['characteristic_index'] = characteristicIndex.toString();
```

Comment:
Could we use an `int` instead of a `String` when serializing and deserializing the index?

```suggestion
    data['characteristic_index'] = characteristicIndex;
```

https://docs.flutter.dev/platform-integration/platform-channels#codec